pageserver/src/tenant/mgr.rs

```diff
@@ -587,7 +587,13 @@ pub(crate) fn tenant_spawn(
         "Cannot load tenant, ignore mark found at {tenant_ignore_mark:?}"
     );
 
-    info!("Attaching tenant {tenant_shard_id}");
+    info!(
+        tenant_id = %tenant_shard_id.tenant_id,
+        shard_id = %tenant_shard_id.shard_slug(),
```

Comment:
Do we need separate tenant and shard id? couldn't we just have one param for `tenant_shard_id`?

Reply:
We did discuss alternatives at least with Christian; we cannot yet use valuable (cfg tracing-unstable) nor would it give us this (selected code fragment) so the other alternative was to create custom macros around all tracing macros, which did not seem too great either, or use a proc macro to do similar which did not seem productive either.

Reply:
The idea of having separate labels is to enable people to confidently filter on tenant_id without having to remember to also search for tenant_shard_id and/or include a shard wildcard -- that convention is new, was established in https://github.com/neondatabase/neon/pull/5957

Reply:
`tenant_shard_id` has both a display and a debug impl, I don't see the problem? This would be similar to [one of my early PRs](https://github.com/neondatabase/neon/pull/4630/files#diff-74152cb7b3681c454c4156d530ce544ee8f15048c9f3cacc736dba5dc7c0d664R4543).

Reply:
> confidently filter on tenant_id

and

> tenant_shard_id has both a display and a debug impl, I don't see the problem?

I think it's the confidence, as in you remain confident to filter logs by:

`tenant_id=XYZ ` vs. `tenant_shard_id=XYZ\S+ ` I guess ... I would not have had a problem.

> This would be similar to [one of my early PRs](https://github.com/neondatabase/neon/pull/4630/files#diff-74152cb7b3681c454c4156d530ce544ee8f15048c9f3cacc736dba5dc7c0d664R4543).

I don't see the connection to there; you were adding a one way for layers to be printed in messages/parameters and using it. Now we use the layer file names everywhere thanks to that :+1:.

Reply:
tbh I never filter logs by `tenant_id=XYZ`, I always just filter by `XYZ`.

Reply:
(but I understand that some use cases might need higher precision than what I am doing, and in those instances being able to filter via `tenant_id=XYZ` makes sense)